libcomposefs/lcfs-writer.c

```diff
@@ -568,17 +592,41 @@ struct lcfs_node_s *lcfs_load_node_from_file(int dirfd, const char *fname,
 	ret->inode.st_size = sb.st_size;
 
 	if ((sb.st_mode & S_IFMT) == S_IFREG) {
-		if (sb.st_size != 0 && (buildflags & LCFS_BUILD_COMPUTE_DIGEST) != 0) {
-			int fd = openat(dirfd, fname, O_RDONLY | O_CLOEXEC);
+		bool compute_digest = (buildflags & LCFS_BUILD_COMPUTE_DIGEST) != 0;
+		bool inline_small = (buildflags & LCFS_BUILD_INLINE_SMALL) != 0;
+		bool do_digest = sb.st_size != 0 && compute_digest;
+		bool do_inline = sb.st_size > 0 && sb.st_size <= 64 && inline_small;
+
+		if (do_digest || do_inline) {
+			cleanup_fd int fd =
+				openat(dirfd, fname, O_RDONLY | O_CLOEXEC);
 			if (fd < 0) {
 				lcfs_node_unref(ret);
 				return NULL;
 			}
-			r = lcfs_node_set_fsverity_from_fd(ret, fd);
-			close(fd);
-			if (r < 0) {
-				lcfs_node_unref(ret);
-				return NULL;
+			if (do_digest) {
+				r = lcfs_node_set_fsverity_from_fd(ret, fd);
+				if (r < 0) {
+					lcfs_node_unref(ret);
+					return NULL;
+				}
+			}
+			if (do_inline) {
+				uint8_t buf[64];
+
+				if (do_digest)
+					lseek(fd, 0, SEEK_SET);
```

Comment:
(Unrelated, maybe `lcfs_node_set_fsverity_from_fd` should be changed to maintain the file descriptor position instead?)

Reply:
I dunno, that would then break use of lcfs_node_set_fsverity_from_fd() with a non-seekable fd (like a pipe).

Reply:
Simpler: why not move this `lseek` underneath the previous `do_digest` conditional?

(I know this is a minor nit, I just wondered how this lseek was related to digests)

Reply:
Well, then we would seek always after set_fsverity, even if we were not gonna inline it... 
Still, maybe less weird code-wise. Lemme do that.